### PR TITLE
Json structure revisions

### PR DESCRIPTION
**Pull request by zhongh**

### CODE REVIEW

**Comment by zhongh** [comment]
@zednis @xgmachina @kolans I have made revisions to the data structure to properly store information on mineral-species, locality, and specific mineral-specimen information, in which the age-determination-process contains how the age is determined. If the age is labelled with "R" (remote), the dated locality is where the age is originally dated. See mapping here: 

https://github.com/tetherless-world/dtdi-mineral-observation-map/blob/json-structure-revisions/ingest/mappings/mineral-observation.json

Below is a screen shot of the browser. It tends to just show all the information on a mineral observation entry in a semi-organised style for discussion at this stage. (We can do further work on how we want the info to be displayed, but for now, let's just display everything first)

screen shot 2016-03-18 at 4 45 47 pm

File: html/browser.html [hunk 1]
```diff
@@ -20,30 +20,90 @@
...
-                    <a href="http://www.google.com/maps/place/{{locality.coordinates}}" target="_blank"> (View in Google Map)</a>
+                    <strong>Locality:</strong>
+                    <a href="{{locality.locality-mindat-url}}" target="_blank">{{locality.locality-label}}</a>
+                    <sup>{{mineral-specimen.age-determination-process.mindat-claims-existence.label}}</sup>
                 </div>
-
-                {{#if locality.min-age}}
-                    {{#if locality.max-age}}
-                        <div><strong>Age:</strong> {{locality.min-age}} - {{locality.max-age}} Ma</div>
+                <div style="font-style: italic; text-indent: 20px">
```
Comment (zednis) [inline html/browser.html:33]: Let's put this style info in a CSS style section and reference it via a named style.  Same below where you have style info in the HTML.

File: ingest/mappings/mineral-observation.json [hunk 2]
```diff
@@ -93,59 +78,115 @@
...
+              "dated-locality-mindat-url":{
+                "type":"string"
+              }
+            }
+          }
+        }
+      },
+      "mineral-species":{
+        "properties":{
+          "chemistry-elements-list":{
```
Comment (zednis) [inline ingest/mappings/mineral-observation.json]: why is chemistry-elements-list a multi_field property?

File: html/browser.html (hunk 1 above)
Comment (zhongh) [inline html/browser.html:33]: Actually I suggest we don't need to implement this now. These style settings are there to make the explanations of the labels more obvious. And we may not show these explanations in every item in the future. I am showing these explanations for now in order to let us have a more clear view of what information we have.
qq20160325-0 2x

File: ingest/mappings/mineral-observation.json (hunk 2 above)
Comment (zhongh) [inline ingest/mappings/mineral-observation.json]: Fixed in the latest commit. Now it's simply:

```
"elements":{
            "type": "string",
            "index": "not_analyzed"
          }
```

Comment (zednis) [inline ingest/mappings/mineral-observation.json]: :+1:

**Comment by zhongh** [comment]
@zednis @xgmachina @kolans After merging Stephan's newest updates on geological times, solving conflicts, and making changes in browser.html and map.html to adapt the newest json data structure, both the browser and the map display look nice and working as below:
qq20160325-1 2x
qq20160325-2 2x

**Comment by zednis** [comment]
Good work @zhonghao1014 !

**Comment by zednis** [comment]
I think there is something wrong with the RRUFF links from the browser search results.  The link generated does not include the mineral name (or any other identifying information).

link: http://rruff.info/mineral_list/locality.php?mineral_name=

screen shot 2016-03-28 at 4 00 40 pm

**Comment by zednis** [comment]
Ah, I just realized I was getting the error because there are mineral observation records in my ES that do not have `mineral-species.mineral-name-plain`.  It is possible this is because I did not wipe my index before importing the restructured JSON.  I will try again and see if a clean index clears the issue up.

**Comment by zednis** [comment]
Re-importing the JSON to a clean index resolved the issue I was having.

**Comment by zednis** [comment]
@zhonghao1014 The refactored JSON looks good to me.

@xgmachina @kolans Do you have any feedback/suggestions on how the age-related information is presented?  (see screenshots from Hao and attached)

screen shot 2016-03-28 at 4 13 13 pm

**Comment by zednis** [comment]
As an alternative we could put the legend for the various superscript symbols (e.g. O, R, etc) below the search results table so we would not have to repeat them.

**Comment by zhongh** [comment]
@zednis Yep, eventually we shall put the legend/explanations of the superscripted label at the bottom of the page. I put it this way for now only because I think we are still not too familiar with the meanings of these labels during development, so I just display them in each result so that one does not need to scroll down the page constantly to see what it means. But, yeah, eventually, we shouldn't show it repeatedly.

**Comment by xgmachina** [comment]
Nice work using the pairs between numerical ages and literal geologic age terms.

File: html/browser.html [hunk 3]
```diff
@@ -20,30 +20,79 @@
     <script id="mineral-observation-template" type="text/x-handlebars-template">
         <tr>
             <td>
-                {{#if mineral.mineral-name-plain}}<div><strong>Mineral Name:</strong> <a href="{{mineral.mineral-mindat-url}}" target="_blank"> {{mineral.mineral-name-plain}}</a></div>
-                {{else}} <div><strong>Mineral Name:</strong> N/A</div>
+                {{#if mineral-species.mineral-name-plain}}<div><strong>Mineral Name:</strong><a href="{{mineral-species.mineral-mindat-url}}" target="_blank">{{mineral-species.mineral-name-plain}}</a></div>
```
Comment (zednis) [inline html/browser.html]: I think we should add back the space between the semi-colon and the mineral name

File: html/browser.html [hunk 4]
```diff
@@ -20,30 +20,79 @@
...
+
+                {{#if mineral-species.rruff-chemistry-html}}
+                    <div>
+                        <strong>Mineral Chemistry:</strong>{{{mineral-species.rruff-chemistry-html}}}
+                    </div>
                 {{/if}}
 
-                {{#if description}}<div><strong>Mineral Description:</strong> {{description}}</div>{{/if}}
+                <div>
+                    <a href="http://rruff.info/mineral_list/locality.php?mineral_name={{mineral-species.mineral-name-plain}}" target="_blank"><strong>RRUFF</strong></a>
```
Comment (zednis) [inline html/browser.html:86]: Let's make this a badge like we have with DCO-ID and DOI in DCO browsers.

Comment (zednis) [inline html/browser.html:86]: We can add the badge in a follow-up branch.

Comment (zhongh) [inline html/browser.html:86]: Sure!

**Comment by zednis** [comment]
I think this branch is ready to merge.  :shipit:

**Comment by zhongh** [comment]
I also think we can close issue #11 and #16

**Comment by zednis** [comment]
fixes #11 fixes #16

**Pull request merged**